File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # define hashset
        hash_set = []
        
        for i in range(0, len(nums)):
            lookingFor = target - nums[i] 
            if lookingFor in hashMap:
                return [hash_set.index(lookingFor), i]
            else:
                hash_set.append(nums[i])
            
        return []

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // define map/dictionary
    let myMap = new Map();
    let outputArray = []
    
    nums.forEach((item, index) => {
        // does target - item exist?
        if (myMap.has(target - item)){
            outputArray = [myMap.get(target - item), index]
        }else {
            myMap.set(item,index)
        }
    })
    
    return outputArray
    
};